tests/sqlparser_mysql.rs

```diff
@@ -507,6 +507,17 @@ fn parse_create_table_comment_character_set() {
     }
 }
 
+#[test]
+fn parse_create_table_gencol() {
+    let sql_default = "CREATE TABLE t1 (a INT, b INT GENERATED ALWAYS AS (a * 2))";
+    let sql_virt = "CREATE TABLE t1 (a INT, b INT GENERATED ALWAYS AS (a * 2) VIRTUAL)";
```

Comment:
What do you think about adding additional support to the AST so you can recover the original SQL --- as implemented now this change will not permit people to know when the original query had `VIRTUAL` or not

Reply:
Virtual is the default, so adding the marker doesn't make a logical difference (at least in the databases I found that support this). Elsewhere in the tests, it looks like you don't preserve differences like this - e.g. `SHOW COLUMNS` vs `SHOW FIELDS` in the MySQL tests. But if you'd like to preserve this one, I'm happy to do that. :slightly_smiling_face: 

If so, I would probably introduce an enum called something like `GenExpressionQualifier` (though that's an ugly name...), which could be None, Virtual or Stored. The Stored option would then be redundant with `GeneratedAs::ExpStored`.

(For my next PR, I'm planning to make the `GENERATED ALWAYS` keywords optional - as they are in MySQL and SQLite. Let me know if you want the presence or absence of those to be preserved too.)

Reply:
> Elsewhere in the tests, it looks like you don't preserve differences like this - e.g. SHOW COLUMNS vs SHOW FIELDS in the MySQL tests. But if you'd like to preserve this one, I'm happy to do that. 🙂

Yes, it is true that this crate is not always consistent. However, for newly added code let's try and preserve the ability to round trip the AST (though I see there is no note describing that detail on https://github.com/sqlparser-rs/sqlparser-rs -- I will try and add a note when I have time)

>  Let me know if you want the presence or absence of those to be preserved too.)

yes, please

Reply:
OK, the latest commit (6d69a43) preserves the distinction between `VIRTUAL`, `STORED` and no qualifier. As I mentioned, that does mean there are two fields which both record if `STORED` was used. That's not so tidy, but it preserves compatibility for any code that's checking for `GeneratedAs::ExpStored`.

I've called the new field & enum `generation_expr_mode` & `GeneratedExpressionMode`, which are both quite a mouthful. Let me know if you have a better name for this. :slightly_smiling_face:

Reply:
filed https://github.com/sqlparser-rs/sqlparser-rs/pull/1052 to document this property